src/components/Dapp.re

```diff
@@ -61,18 +61,26 @@ module Streak = {
 module DisplayAfterDate = {
   [@react.component]
   let make = (~endDateMoment, ~beforeComponent, ~afterComponent) => {
-    let isBeforeDate = () =>
-      MomentRe.diff(endDateMoment, MomentRe.momentNow(), `seconds) > 0.;
+    let isBeforeDate =
+      React.useCallback1(
+        () =>
+          MomentRe.diff(endDateMoment, MomentRe.momentNow(), `seconds) > 0.,
+        [|endDateMoment|],
+      );
 
     let (beforeDate, setIsBeforeDate) = React.useState(() => isBeforeDate());
-    React.useEffect(() => {
-      let _timout =
-        Js.Global.setTimeout(
-          () => setIsBeforeDate(_ => isBeforeDate()),
-          1500,
-        );
-      None;
-    });
+    React.useEffect2(
+      () => {
+        Js.log("this was called!!!");
+        let timeout =
+          Js.Global.setTimeout(
+            () => setIsBeforeDate(_ => isBeforeDate()),
+            1500,
+          );
+        Some(() => Js.Global.clearTimeout(timeout));
+      },
+      (setIsBeforeDate, isBeforeDate),
```

Comment:
I've added dependencies like this to many of the `useEffect` functions. It was sloppy that I wasn't before.